File: Data Structures & Algorithms/meeting-schedule/submission-5.js
/**
 * Definition of Interval:
 * class Interval {
 *   constructor(start, end) {
 *     this.start = start;
 *     this.end = end;
 *   }
 * }
 */

class Solution {
    /**
     * @param {Interval[]} intervals
     * @returns {boolean}
     */
    canAttendMeetings(intervals) { 
        intervals.sort((a, b) => a.start - b.start) 
        console.log(intervals)
        for (let i = 0; i < intervals.length - 1; i ++) {
            let cur = intervals[i] 
            let nxt = intervals[i + 1]  

            if (cur.end > nxt.start) {
                return false
            }
        } 

        return true
    }
}

File: Data Structures & Algorithms/insert-new-interval/submission-5.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        intervals.append(newInterval) 
        intervals.sort(key= lambda x: x[0]) 

        merged = [intervals[0]] 

        for i in range(1, len(intervals)): 
            start, end = intervals[i] 
            lastMerged = merged[-1] 
            
            if start <= lastMerged[1]: 
                lastMerged[1] = max(lastMerged[1], end) 
            else: 
                merged.append([start, end]) 

        return merged
